### ISSUE TITLE
[Godot] Transparent effects revealed for zero mix duration animations

### ISSUE THREAD

**Issue by LemonadeFlashbang** [EsotericSoftware/spine-runtimes#2326]
Hello,

Quite possible this is user error (I'm hoping it is), but I'm unsure about how to get this to work.

I have a character with the following animation:

![azla_working](https://github.com/EsotericSoftware/spine-runtimes/assets/81896582/95f4b71d-049f-47c5-b945-235b4285dfc0)

It's a karate chop animation with a black trail. When run from the editor or from stationary, this animation works as intended.

However, in-game the effect trail plays twice- like this.

![azla_glitch](https://github.com/EsotericSoftware/spine-runtimes/assets/81896582/9071e0b5-ed26-4fc1-a9a0-6df362c85e47)

After a fair amount of experimentation, it appears this is an artifact of mix times. The "idle" frame blends into the "attack" frame with a mix time of 0.2- and even though that effect doesn't have a keyframe for idle I guess it's defaulting to "alpha = 1." The net result is the effect flashes temporarily before reappearing when intended.

I thought this might be a straightforward solve, and went into the editor and manually configured some mixes. The runtime API states I can mix between two specific animations, but on Godot 3.x the editor only allows mixing for individual animations:

![spinemixes](https://github.com/EsotericSoftware/spine-runtimes/assets/81896582/e6b57fa0-0886-4924-b3a7-9ee2c2a95003)

Setting the mix for attack to zero has no effect. Neither does setting the mix for idle. There shouldn't _be_ any mixing, but it still appears like the alpha is being mixed in and the visual is appearing.

Disabling mix time altogether isn't ideal- I need it for the other animations. It's not clear to me how I should be configuring mix time for these parameters.

So to summarize:
1) Setting any mix time causes effects with zero alpha to become visible.
2) Godot's animation mixes on 3.x don't seem to allow me to remove mix times between two animations. I recompiled my engine using this repo only about 2-3 weeks ago, so I should be on the latest version.
3) There doesn't appear to be a way to remove the mix time before a specific animation.

Fixing any of these should hopefully resolve my issue, but thought I'd document all three as  being pain points for me.

**Comment by LemonadeFlashbang** [comment]
And here's a reproducible project. Change to any animation with an arrow key, then back to the attack animation using the "up" key. Both the "idle" and "attack" animations have a mix of 0- but going from idle -> attack will still cause the issue even though there should be no blending.

[Spine Mixing Repro Project.zip](https://github.com/EsotericSoftware/spine-runtimes/files/12188002/Spine.Mixing.Repro.Project.zip)

**Comment by badlogic** [comment]
Thanks for the repro. Could you tell me what you mean by "alpha"? I'm also a little confused as your "animation mixes" panel only shows a single animation name. However, animation mixes are always defined between two animations. 

<img width="580" alt="Screenshot 2023-07-28 at 22 31 43" src="https://github.com/EsotericSoftware/spine-runtimes/assets/514052/6e4cc79b-b010-49d8-84ed-a04e142ca147">

Here's what I get when I expand the inspector. The mix times are defined as:

* "attack" -> "attack": 0
* "attack" -> "idle": 0

You also have a default mix time of 0.2 seconds. This will be used for any animation pair for which you have not set a mix time explicitely. E.g. "idle" -> "attack" will have mix time 0.2 because of this.

Add another animation mix for "idle" -> "attack" and set it to 0, that should fix your issue.

**Comment by LemonadeFlashbang** [comment]
I don't have the option to add another animation to the mix, and I haven't figured out a way to add them programmatically because I can't seem to retrieve the AnimationStateData object. The screenshot is exactly what appears in my version of Godot. 

I just reconstructed the editor too, so I don't believe I'm on an out of date version.

![image](https://github.com/EsotericSoftware/spine-runtimes/assets/81896582/c02bd496-631e-44ba-b7cc-0e1784085b89)


Reproduction steps:

1. Head to the [Spine Runtimes Guide](http://en.esotericsoftware.com/spine-godot#Installation)
2. Clone the Spine Runtimes repository (this one)
3. Run the setup.sh script targeting 3.5. I am using this [custom fork ](https://github.com/LemonadeFlashbang/godot/tree/3.5)which is identical save for a text processing fix.
4. Run:` ./build/build.sh release_debug`
5. Open the project.

I've hosted a fork of this repo that targets my custom Godot fork by default as well, [which you can find here](https://github.com/LemonadeFlashbang/spine-runtimes), but again it's all almost identical. The ability to set mix states between animations is simply not present.

As for alpha, I'm talking about the color. Here's they Keyframe in the Spine editor:

![image](https://github.com/EsotericSoftware/spine-runtimes/assets/81896582/e8d1892c-e495-464b-aa01-8050dfc61158)
The very first frame sets the "alpha" (transparency) of the effect to zero. It remains that way for 15 frames, after which it becomes visible and fades out.

This same effect has _no_ keyframes in the idle animation. The behavior I believe is happening is that when blending the two animations, it treats "no keyframe" as being fully visible and then blends that into the new animation when it shouldn't.

**Comment by badlogic** [comment]
You can add another mix by clicking the "Add mix" button that you can see in both our screen shots.

Please make the inspector view bigger so you can see both the from and to animations for each mix. Then add a mix for "idle" -> "attack" and set it to 0.

**Comment by LemonadeFlashbang** [comment]
Oh. It's just an inspector size issue. I probably would never have tried that, so thank you.

That's a bit embarrassing, but hopefully someone else can learn from this.

**Comment by badlogic** [comment]
No worries, the Godot UI sadly doesn't indicate that there is more to see.
